Remove native_functions.yaml dependency from CUDA distributions

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73162
* __->__ #67875

Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

peterbell10 committed:
Remove native_functions.yaml dependency from CUDA distributions

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"




[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"




[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from CUDA distributions"




[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

dagitses commented on aten/src/ATen/native/cuda/Distributions.cpp:
```diff
@@ -0,0 +1,72 @@
+namespace at { namespace native {
+
+Tensor _s_poisson_cuda(const Tensor& lambda, c10::optional<Generator> gen_) {
+  auto gen = get_generator_or_default<CUDAGeneratorImpl>(gen_, cuda::detail::getDefaultCUDAGenerator());
```
these calls are flagging:
> Call to at::cuda::detail::getDefaultCUDAGenerator is logically private, please avoid use outside of its parent namespace

Do you think this is bogus or is there any refactoring that makes sense here?

peterbell10 replied:
It sounds like it doesn't like that a `detail` namespace is being used outside of the `at::cuda` namespace but that's just how ATen is structured.

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

Commit:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

Commit:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

Commit:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

dagitses commented:
@dagitses has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Rebase and fix merge conflicts on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Fix overflow for 32-bit size_t on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Rebase on master on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 commented:
@malfet ping

malfet commented:
@peterbell10 this change break a bunch of Windows builds are `struct` and `class` are not interchangeable by VC++, let me try to rebase your change and see if it already fixed the problem, if not I'll suggest a small fix

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32500425).

peterbell10 committed:
Rebase and fix forward-declarations on "Remove native_functions.yaml dependency from CUDA distributions"


Differential Revision: [D32500425](https://our.internmc.facebook.com/intern/diff/D32500425)

[ghstack-poisoned]

peterbell10 commented:
I think I got it. Weird that open source CI didn't catch it though.

malfet commented:
> I think I got it. Weird that open source CI didn't catch it though.

Yes, I believe this is it, giving PR another try
